lighthouse-core/lib/network-request.js

```diff
@@ -341,10 +363,89 @@ module.exports = class NetworkRequest {
     if (!totalFetchedSize) return;
-    this.transferSize = parseFloat(totalFetchedSize.value);
+    const floatValue = parseFloat(totalFetchedSize.value);
+    // Bail if the header cannot be parsed
+    if (isNaN(floatValue)) return;
+    this.transferSize = floatValue;
+  }
+
+  /**
+   * TODO(exterkamp) add explanatory comment.
+   */
+  _updateTimingsForLightrider() {
+    // Bail if we aren't in Lightrider.
+    if (!global.isLightrider) return;
+
+    // Bail if timing is not initialized.
+    if (!this.timing) {
+      return;
+    }
+
+    // For more info on timing nomenclature: https://www.w3.org/TR/resource-timing-2/#processing-model
+
+    //    StartTime
+    //    | ConnectStart
+    //    | |     SSLStart  SSLEnd
+    //    | |     |         | ConnectEnd
+    //    | |     |         | | SendStart/End   ReceiveHeadersEnd
+    //    | |     |         | | |               |                EndTime
+    //    ▼ ▼     ▼         ▼ ▼ ▼               ▼                ▼
+    //    [ [TCP  [   SSL   ] ] [   Request   ] [   Response   ] ]
+    //    ▲ ▲     ▲         ▲ ▲ ▲             ▲ ▲              ▲ ▲
+    //    | |     '-SSLMs---' | '-requestMs---' '-responseMs---' |
+    //    | '----TCPMs--------'                                  |
+    //    |                                                      |
+    //    '------------------------TotalMs-----------------------'
+
+    const totalHeader = this.responseHeaders.find(item => item.name === HEADER_TOTAL);
+    // Bail if there was no totalTime.
+    if (!totalHeader) return;
+
+    const totalMs = parseInt(totalHeader.value);
+
+    const TCPHeader = this.responseHeaders.find(item => item.name === HEADER_TCP);
+    const requestHeader = this.responseHeaders.find(item => item.name === HEADER_REQ);
+    const SSLHeader = this.responseHeaders.find(item => item.name === HEADER_SSL);
+    const responseHeader = this.responseHeaders.find(item => item.name === HEADER_RES);
+
+    // Make sure all Times are initialized and are non-negative.
+    const TCPMs = Math.max(0, TCPHeader === undefined ? 0 : parseInt(TCPHeader.value));
+    const requestMs = Math.max(0, requestHeader === undefined ? 0 : parseInt(requestHeader.value));
+    const SSLMs = Math.max(0, SSLHeader === undefined ? 0 : parseInt(SSLHeader.value));
+    const responseMs = Math.max(0,
+      responseHeader === undefined ? 0 : parseInt(responseHeader.value));
+
+    // Bail if the timings don't add up.
+    if (TCPMs + requestMs + responseMs !== totalMs) {
+      return;
+    }
+
+    const origEnd = this.endTime;
+
+    // EndTime and responseReceivedTime are in seconds, so conversion is necessary.
+    this.endTime = this.startTime + (totalMs / 1000);
+    this.responseReceivedTime = this.startTime + ((TCPMs + requestMs) / 1000);
+
+    this.timing.connectStart = 0;
+    this.timing.connectEnd = TCPMs;
+    // Make sure that SSL time is less than or equal to total TCP time.
+    if (SSLMs <= TCPMs) {
```

Comment:
is it a bad sign for other things? Should we bail if this isn't true?

Reply:
tbh I have no idea.  I don't think this will ever happen.  If it does I don't know.  

We can assume it does and bail? :man_shrugging:

Reply:
this wont ever happen based on LR fetching code. we can bail hard on this condition.